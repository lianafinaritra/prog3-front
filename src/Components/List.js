import Card from "./Card";

export default function List( props ) {
    const { handleUpdate, tab, cardStatus } = props;

    return(
        <div className="card-container" data-testid="list">
            {
                tab.filter(item => item.status === cardStatus).map((elt, k)=>(
                    <Card item={elt} handleUpdate={handleUpdate} key={k} />
                ))
            }
        </div>
    )
}