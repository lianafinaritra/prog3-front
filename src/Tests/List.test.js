import { render } from "@testing-library/react";
import React, {useState} from "react";
import List from "../Components/List";
import renderer from "react-test-renderer";

const tabEmpty = [];

const tab = [
    {
        id: 1,
        text : "enjana",
        status : "TODO",
    }
]

describe('List test', () => {
    const handleUpdate = jest.fn();

    it('renders list elements', () => {
        const { getByTestId } = render(<List handleUpdate={handleUpdate} tab={tab} cardStatus="TODO" />);
        expect(getByTestId('list')).toBeInTheDocument();
        expect(getByTestId('cardItem')).toBeInTheDocument();
        expect(getByTestId('cardItem').textContent).toBe('enjana');
    });

    it('render anything when tab is empty', () => {
        const{ container, getByTestId } = render(
            <List handleUpdate={handleUpdate} tab={tabEmpty} cardStatus="TODO" />
        )
        expect(getByTestId("list")).toBeTruthy();
        expect(container.querySelector('.card')).toBeNull();
    });

    it('render list card when tab have item', () => {
        const{ getByTestId } = render(
            <List handleUpdate={handleUpdate} tab={tab} cardStatus="TODO" />
        )
        expect(getByTestId("list")).toBeTruthy();
        expect(getByTestId("cardItem")).toBeTruthy();
    });

    it("should renders correctly", ()=>{
        const listSnap = renderer.create(
            <List handleUpdate={handleUpdate} tab={tab} cardStatus="TODO" />
        )

        const list = listSnap.toJSON();
        expect(list).toMatchSnapshot();
    })

});

