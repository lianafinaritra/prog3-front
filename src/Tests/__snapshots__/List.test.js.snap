// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test should renders correctly 1`] = `
<div
  className="card-container"
  data-testid="list"
>
  <div
    className="card"
    data-testid="cardItem"
  >
    <p>
      enjana
    </p>
    <div
      className="checkbox"
    >
      <div
        data-testid="checkbox"
      >
        <input
          onClick={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;
